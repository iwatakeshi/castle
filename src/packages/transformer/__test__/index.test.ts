// import { parse, print, types } from 'recast'
// import { namedTypes, NamedTypes } from 'ast-types/gen/namedTypes'
// import { customTsParser } from '..'
// import { ASTNode } from 'ast-types'
// import { builders } from 'ast-types/gen/builders'

import transform from '..'
import j from 'jscodeshift'

describe('packages/transformer', () => {
  it('it should return an identity (ast -> ast)', () => {
    const file = `const a = 1`
    const code = transform(file, function (ast) {
      return ast
    })

    expect(code).toEqual(file)
  })

  it('it should return const b = 1', () => {
    const file = `const a = 1`
    const code = transform(file, function (program) {
      program.find(j.Identifier).forEach(path => {
        j(path).replaceWith(
          j.identifier('b')
        );
      })
      return program
    })

    expect(code).toEqual(`const b = 1`)
  })
})
