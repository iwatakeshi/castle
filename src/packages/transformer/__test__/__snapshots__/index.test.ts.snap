// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`packages/transformer should add import at start of file with no imports present 1`] = `
"import React from \\"react\\";
export const truth = () => 42"
`;

exports[`packages/transformer should add import at the end of all imports if imports are present 1`] = `
"import React from 'react'
import \\"app/styles/app.css\\";
export default function Comp() {
  return <div>hello world!</div>
}"
`;
